Add VizPanelExploreButton scene object

domasx2 opened the pull request:
Hey,

On popular request, contributing a panel explore button derived from app o11y implementation. It's been copied around to other apps, so seems to make sense to contribute it to scenes.

Does this make sense? If so, I'll add tests. Also thinking it's maybe whorthwile to make a `getExploreUrl(sceneObject)` util as it's own thing in case you want something else than a button

![panel explore button](https://github.com/grafana/scenes/assets/847684/37158277-c794-4edd-a2e0-ae6fdb69e301)

domasx2 commented on packages/scenes/src/components/VizPanel/VizPanelExploreButton.tsx:
```diff
@@ -0,0 +1,91 @@
+  }
+}
+
+function VizPanelExploreButtonComponent({ model }: SceneComponentProps<VizPanelExploreButton>) {
+  const { options } = model.useState();
+
+  const { data } = sceneGraph.getData(model).useState();
+
+  const targets = useMemo(() => data?.request?.targets ?? [], [data]);
+
+  const { from, to } = sceneGraph.getTimeRange(model).useState();
+
+  const { value: interpolatedQueries } = useAsync(async () => {
+    const scopedVars: ScopedVars = {
+      __sceneObject: { text: '__sceneObject', value: model },
+    };
+    return (
+      await Promise.allSettled(
+        targets.map(async (q) => {
+          const queryDs = await getDataSourceSrv().get(q.datasource);
+          return queryDs.interpolateVariablesInQueries?.([q], scopedVars ?? {})[0] || q;
+        })
+      )
+    )
+      .filter((promise): promise is PromiseFulfilledResult<DataQuery> => promise.status === 'fulfilled')
+      .map((q) => q.value)
+      .map((q) => options.transform?.(q) ?? q);
+  }, [targets, model]);
+
+  const left = useMemo(() => {
```
I think this button is a link styled as a button, it does have proper `href`. the `onClick` is an optional callback for adding tracking of clicks via Faro/Rudderstack if needed. However I'm now not sure it will react to variable updates properly, needs investigation :thinking: maybe onHover is a good idea

torkelo commented:
@domasx2 why did you close, think this is a great addition.  

sorry for not approving it, it must have slipped of my radar / review list

domasx2 commented:
I realized this has been hanging for a long while, and thought if I didn't find time and energy to push it, best close it. Do you think with some tests added this would be ok to merge, or needs more work?

domasx2 commented on packages/scenes/src/components/VizPanel/VizPanelExploreButton.tsx:
```diff
@@ -0,0 +1,91 @@
+
+  const { data } = sceneGraph.getData(model).useState();
+
+  const targets = useMemo(() => data?.request?.targets ?? [], [data]);
```
done

domasx2 commented:
extracted "getExploreURL" as separate util for use cases outside of button component, added "return to previous" integration, made it ad hoc filters aware
